### PR TITLE
Enable config verification and fix compilation with lapack

### PR DESCRIPTION
**Pull request by mshabunin**
Checklist
* [x] Used a fork of the feedstock to propose changes
* [x] Bumped the build number (if the version is unchanged)
* [x] Reset the build number to `0` (if the version changed)
* [x] [Re-rendered]( https://conda-forge.org/docs/maintainer/updating_pkgs.html#rerendering-feedstocks ) with the latest `conda-smithy` (Use the phrase <code>@<space/>conda-forge-admin, please rerender</code> in a comment in this PR for automated rerendering)
* [x] Ensured the license file is being packaged.

`ENABLE_CONFIG_VERIFICATION` option will ensure each requested cmake `WITH_*` option is consistent with actual configuration.

### CODE REVIEW

**Comment by mshabunin** [comment]
@conda-forge-admin, please rerender

File: recipe/build.sh [hunk 1]
```diff
@@ -90,8 +90,7 @@ cmake -LAH -G "Ninja"                                                     \
     -DENABLE_CONFIG_VERIFICATION=ON                                       \
     $CPU_DISPATCH_FLAGS                                                   \
     $OPENMP                                                               \
-    -DLAPACK_LIBRARIES="-llapacke -llapack -lcblas -lblas"                \
-    -DWITH_LAPACK=1                                                       \
```
Comment (hmaarrfk) [inline recipe/build.sh]: I think the intent was to compile with lapacke by @isuruf 

That said, if it was "always broken" I'm ok Merging this so we can move forward with the other stuff.

File: recipe/meta.yaml [hunk 2]
```diff
@@ -48,7 +48,6 @@ requirements:
     - libpng
     - ffmpeg 4.1.*                   # [not win]
     - qt                             # [not osx]
-    - liblapacke
```
Comment (hmaarrfk) [inline recipe/meta.yaml]: My initial thought is to keep liblapcke as a dependency even if we don't use it.

This will help the solver avoid selecting this specific version of opencv to avoid a lapacke dependency while we add lalack back in.

**Comment by isuruf** [comment]
@conda-forge-admin, rerender

File: recipe/build.sh [hunk 3]
```diff
@@ -87,9 +88,9 @@ cmake -LAH -G "Ninja"                                                     \
     -DCMAKE_INSTALL_LIBDIR="lib"                                          \
     $CMAKE_TOOLCHAIN_CMD_FLAGS                                            \
     -DOPENCV_GENERATE_PKGCONFIG=ON                                        \
+    -DENABLE_CONFIG_VERIFICATION=ON                                       \
     $CPU_DISPATCH_FLAGS                                                   \
     $OPENMP                                                               \
-    -DLAPACK_LIBRARIES="-llapacke -llapack -lcblas -lblas"                \
```
Comment (isuruf) [inline recipe/build.sh]: Why is this removed?

File: recipe/build.sh [hunk 4]
```diff
@@ -79,6 +79,7 @@ PYTHON_UNSET_INSTALL="-DOPENCV_PYTHON${PY_UNSET_MAJOR}_INSTALL_PATH=${SP_DIR}"
 
 # FFMPEG building requires pkgconfig
 export PKG_CONFIG_PATH=$PKG_CONFIG_PATH:$PREFIX/lib/pkgconfig
+export OpenBLAS_HOME=$PREFIX
```
Comment (isuruf) [inline recipe/build.sh]: This should be removed

**Comment by hmaarrfk** [comment]
@conda-forge-admin, please rerender

**Comment by hmaarrfk** [comment]
@isuruf how can we confirm if opencv was actually being compiled with lapacke? the build process seems to think it isn't for windows.

File: recipe/build.sh (hunk 4 above)
Comment (mshabunin) [reply recipe/build.sh]: @isuruf , we are targeting OpenBLAS case here: 

https://github.com/opencv/opencv/blob/b2abd8ca4197744174d34d96e3053f8e0d7fad54/cmake/OpenCVFindLAPACK.cmake#L89-L98

which will call this file:

https://github.com/opencv/opencv/blob/master/cmake/OpenCVFindOpenBLAS.cmake

which uses environment variable `OpenBLAS_HOME` or hardcoded paths for detection.

**Comment by mshabunin** [comment]
@conda-forge-admin, please rerender

**Comment by mshabunin** [comment]
@conda-forge-admin, please rerender

**Comment by isuruf** [comment]
We don't directly link to openblas anymore. Have a look at https://conda-forge.org/docs/maintainer/knowledge_base.html#blas

**Comment by isuruf** [comment]
and https://conda-forge.org/docs/maintainer/knowledge_base.html#how-it-works

**Comment by mshabunin** [comment]
@conda-forge-admin, please rerender

**Comment by mshabunin** [comment]
@isuruf , thanks, that was very useful.

I hope the PR is ready.

BTW, is it possible to check resulting packages with different LAPACK implementations (openblas, mkl) during the test stage?

**Comment by hmaarrfk** [comment]
> BTW, is it possible to check resulting packages with different LAPACK implementations (openblas, mkl) during the test stage?

You could try adding a dependency per python version.

Say python 3.6 tests with openblas, and python 3.7 tests with MKL. I think that should be good enough :/

File: recipe/run_test.py [hunk 5]
```diff
@@ -28,7 +30,19 @@ def tearDownClass(cls):
...
+        self.assertTrue(res, "Can not read video frame from file")
+
+
+class TestGEMM(unittest.TestCase):
+    def test_gemm_big(self):
+        sz = (500, 500)
+        a = np.ones(sz, dtype=float)
+        b = np.eye(sz[0])
+        c = np.ones(sz, dtype=float)
+        x = cv2.gemm(a, b, 2, c, 3)
```
[the remark this answers is not in the document]
Comment (mshabunin) [reply recipe/run_test.py]: No, it should not, but it can raise an error if LAPACK integration is broken (e.g. corrupting memory).

**Comment by mshabunin** [comment]
> Say python 3.6 tests with openblas, and python 3.7 tests with MKL

It works, thank you.

**Comment by hmaarrfk** [comment]
@conda-forge-admin, please

**Comment by mshabunin** [comment]
@hmaarrfk , builds fail randomly because of internet connection issues. cmake can not download IPPICV, this feature got disabled and config verification fails. I'll try to introduce a workaround. Do you know whether curl or wget is available on builders?

**Comment by mshabunin** [comment]
Similar problem with other dependencies, e.g. FFmpeg plugin on Windows:
Download fails:
```
-- FFMPEG: Download: opencv_ffmpeg.dll
-- FFMPEG: Download: opencv_ffmpeg_64.dll
CMake Warning at cmake/OpenCVDownload.cmake:193 (message):
  FFMPEG: Download failed: 22;"HTTP response code said error"

  For details please refer to the download log file:

  D:/bld/opencv_1558353242981/work/build/CMakeDownloadLog.txt

Call Stack (most recent call first):
  3rdparty/ffmpeg/ffmpeg.cmake:20 (ocv_download)
  modules/videoio/cmake/detect_ffmpeg.cmake:14 (download_win_ffmpeg)
  modules/videoio/cmake/init.cmake:3 (include)
  modules/videoio/cmake/init.cmake:24 (add_backend)
  cmake/OpenCVModule.cmake:312 (include)
  cmake/OpenCVModule.cmake:375 (_add_modules_1)
  modules/CMakeLists.txt:7 (ocv_glob_modules)


-- FFMPEG: Download: ffmpeg_version.cmake
```
Verification fails:
```
CMake Error at cmake/OpenCVUtils.cmake:695 (message):
  Some dependencies have not been found or have been forced, unset
  ENABLE_CONFIG_VERIFICATION option to ignore these failures or change
  following options:

  WITH_FFMPEG
Call Stack (most recent call first):
  CMakeLists.txt:1656 (ocv_verify_config)
```

**Comment by hmaarrfk** [comment]
i think curl is available. you can depend on it if it isn't.

**Comment by isuruf** [comment]
On windows, opencv is trying to downloading ffmpeg from the internet instead of using the conda package?

**Comment by mshabunin** [comment]
@isuruf , yes, this is [default behavior](https://github.com/opencv/opencv/blob/de977cc9c836dd954bc0a737b5aa80621a4f9c72/modules/videoio/cmake/detect_ffmpeg.cmake#L12-L25) on Windows. There exists [cmake package support](https://github.com/opencv/opencv/blob/de977cc9c836dd954bc0a737b5aa80621a4f9c72/modules/videoio/cmake/detect_ffmpeg.cmake#L2-L10) but I haven't try it yet. I'll take a look.

**Comment by hmaarrfk** [comment]
Well this isn't any worse than before is it? in fact, at least we know if opencv didn't pull in a dependency. If that dynamic download is the only issue, I think we should pull this in, and work on the rest of the stuff in a separate PR.

**Comment by mshabunin** [comment]
@hmaarrfk , yes, it is not worse from the package perspective. But unstable CI will make future contributions a lot more problematic. I'll try to introduce a workaround today or tomorrow.

FFmpeg on Windows can be postponed.

**Comment by hmaarrfk** [comment]
@conda-forge-admin please rerender

**Comment by hmaarrfk** [comment]
Awesome. This is a great step!

**Comment by mshabunin** [comment]
Thank you!

Win64_Py37 CI job demonstrated cmake download workarounds in action, I hope current parameters (5 tries, 30 sec conection timeout, 3 min download timeout) will be enough for most cases.
```
2019-05-26T19:16:44.2079558Z -- IPPICV: Download: ippicv_2019_win_intel64_20180723_general.zip
2019-05-26T19:17:14.6860672Z -- Try 1 failed
2019-05-26T19:17:45.0731371Z -- Try 2 failed
2019-05-26T19:17:45.0810274Z -- [download 0% complete]
2019-05-26T19:17:45.0902425Z -- [download 1% complete]
...
2019-05-26T19:17:45.3011158Z -- [download 96% complete]
2019-05-26T19:17:45.3053966Z -- [download 100% complete]
```
